File: back-end/src/main/java/nl/hva/backend/domain/models/game/action/impl/card/CardBidAction.java
package nl.hva.backend.domain.models.game.action.impl.card;

import lombok.NoArgsConstructor;
import nl.hva.backend.domain.models.game.Player;
import nl.hva.backend.domain.models.game.action.impl.gamePlay.GameAssetBidAction;
import nl.hva.backend.domain.models.game.card.impl.JailBreakCard;

import javax.persistence.Entity;

@Entity
@NoArgsConstructor
public class CardBidAction extends GameAssetBidAction {

    public CardBidAction(
            Player actor,
            JailBreakCard.JailBreakCardBid bid
    ) {
        super(actor, bid);
    }

    @Override
    public JailBreakCard.JailBreakCardBid getBid() {
        return (JailBreakCard.JailBreakCardBid) super.getBid();
    }

}

File: front-end/src/app/domain/models/game/action/impl/card/CardBidAction.ts
import {GameAssetBidAction} from "../gamePlay/GameAssetBidAction";
import {Player} from "../../../Player";
import {JailBreakCardBid} from "../../../card/impl/JailBreakCardBid";

/**
 * TODO: JsDoc
 *
 * @author Hamza el Haouti
 */
export class CardBidAction extends GameAssetBidAction {

  public constructor(
    id: number,
    actor: Player,
    performedAt: Date,
    bid: JailBreakCardBid
  ) {
    super(CardBidAction.name, id, actor, performedAt, bid);
  }

  override get bid(): JailBreakCardBid {
    return <JailBreakCardBid>super.bid;
  }

  public static trueCopy(bid: CardBidAction): CardBidAction {
    return new CardBidAction(
      bid.id,
      Player.trueCopy(bid.actor),
      new Date(bid.performedAt),
      JailBreakCardBid.trueCopy(bid.bid),
    );
  }

}
